crates/bevy_render/src/render_graph/node.rs

```diff
@@ -131,6 +132,7 @@ impl NodeState {
             input_slots: ResourceSlots::from(node.input()),
             output_slots: ResourceSlots::from(node.output()),
             node: Box::new(node),
+            type_name: Some(std::any::type_name::<T>()),
```

Comment:
Does it always have a value? Is the `Option` needed?

Reply:
In bevy all uses of `NodeState` supply the type name, so the option isn’t needed. However I wasn’t sure whether there are use cases where the type name is not available, and the size of Option<&str> is the same as &str anyway.

I don’t really care which one it is.

Reply:
its less about size and more about adding another branch to unwrap the option (which has performance and code complexity implications). I think I'd rather opt for non-options first, then add them if/when we hit a case where they're needed

Reply:
I changed it now.